pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperatorTest.java

```diff
@@ -24,98 +24,222 @@
+    Assert.assertEquals(resultBlock2.getContainer().get(1), new Object[]{"foo", 4});
+    Assert.assertEquals(resultBlock3.getContainer().size(), 0);
     Assert.assertTrue(operator.nextBlock().isEndOfStreamBlock(), "Expected EOS after reading two rows");
   }
 
   @Test
   public void shouldGetErrorBlockWhenInstanceResponseContainsError() {
     // Given:
+    QueryContext queryContext = QueryContextConverterUtils.getQueryContext("SELECT strCol, intCol FROM tbl");
     DataSchema schema = new DataSchema(new String[]{"strCol", "intCol"},
         new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.STRING, DataSchema.ColumnDataType.INT});
     InstanceResponseBlock errorBlock = new InstanceResponseBlock();
     errorBlock.addException(QueryException.QUERY_EXECUTION_ERROR.getErrorCode(), "foobar");
     List<InstanceResponseBlock> resultsBlockList = Arrays.asList(
         new InstanceResponseBlock(new SelectionResultsBlock(schema,
-            Arrays.asList(new Object[]{"foo", 1}, new Object[]{"", 2})), null),
+            Arrays.asList(new Object[]{"foo", 1}, new Object[]{"", 2})), queryContext),
         errorBlock,
-        new InstanceResponseBlock(new SelectionResultsBlock(schema, Collections.emptyList()), null));
+        new InstanceResponseBlock(new SelectionResultsBlock(schema, Collections.emptyList()), queryContext));
+    LeafStageTransferableBlockOperator operator = new LeafStageTransferableBlockOperator(resultsBlockList, schema);
+
+    // When:
+    TransferableBlock resultBlock = operator.nextBlock();
+
+    // Then:
+    Assert.assertTrue(resultBlock.isErrorBlock());
+  }
+
+  @Test
+  public void shouldReorderWhenQueryContextAskForNotInOrderGroupByAsDistinct() {
+    // Given:
+    QueryContext queryContext = QueryContextConverterUtils.getQueryContext(
+        "SELECT intCol, strCol FROM tbl GROUP BY strCol, intCol");
+    // result schema doesn't match with DISTINCT columns using GROUP BY.
+    DataSchema schema = new DataSchema(new String[]{"intCol", "strCol"},
+        new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.INT, DataSchema.ColumnDataType.STRING});
+    List<InstanceResponseBlock> resultsBlockList = Collections.singletonList(
+        new InstanceResponseBlock(new DistinctResultsBlock(mock(DistinctAggregationFunction.class),
+            new DistinctTable(schema, Arrays.asList(
+                new Record(new Object[]{1, "foo"}), new Record(new Object[]{2, "bar"})))), queryContext));
+    LeafStageTransferableBlockOperator operator = new LeafStageTransferableBlockOperator(resultsBlockList, schema);
+
+    // When:
+    TransferableBlock resultBlock = operator.nextBlock();
+
+    // Then:
+    Assert.assertEquals(resultBlock.getContainer().get(0), new Object[]{1, "foo"});
+    Assert.assertEquals(resultBlock.getContainer().get(1), new Object[]{2, "bar"});
+  }
+
+  @Test
+  public void shouldParsedBlocksSuccessfullyWithDistinctQuery() {
```

Comment:
Should we also test the code path for composeColumnIndexedTransferableBlock and composeAggregationTransferableBlock

Reply:
also tested in shouldHandleDesiredDataSchemaConversionCorrectly